src/Ldap/Client/Internal.hs

```diff
@@ -50,10 +52,12 @@ data Host =
     deriving (Show, Eq, Ord)
 
 -- | A token. All functions that interact with the Directory require one.
-data Ldap = Ldap
+data Ldap b = Ldap
```

Comment:
Can we avoid tracking the type of the helper threads' results? They never return anything interesting.